src/Approximations/overapproximate.jl

```diff
@@ -1736,3 +1736,50 @@ function overapproximate(CHA::ConvexHullArray{N, <:AbstractZonotope{N}}, ::Type{
         return Zaux
     end
 end
+
+"""
+    overapproximate(Z::AbstractZonotope, ::Type{<:HParallelotope}, indices=1:dim(Z))
+
+Overapproximation of a zonotopic set with a parallelotopic set in constraint
+representation.
+
+### Input
+
+- `Z`              -- zonotopic set
+- `HParallelotope` -- type for dispatch
+- `indices`        -- (optional; default: `1:dim(Z)`) generator indices selected
+                       when constructing the parallelotope
+
+### Output
+
+An overapproximation of the given zonotope using a parallelotope.
+
+### Algorithm
+
+The algorithm is based on Proposition 8 discussed in Section 5 of [1].
+
+[1] Althoff, M., Stursberg, O., & Buss, M. (2010). *Computing reachable sets of
+hybrid systems using a combination of zonotopes and polytopes*. Nonlinear
+analysis: hybrid systems, 4(2), 233-249.
+"""
+function overapproximate(Z::AbstractZonotope, ::Type{<:HParallelotope}, indices=1:dim(Z))
+    Zred = _overapproximate_hparallelotope(Z, indices)
+    return convert(HParallelotope, Zred)
+end
+
+function _overapproximate_hparallelotope(Z::AbstractZonotope, indices=1:dim(Z))
+    length(indices) == dim(Z) || throw(ArgumentError("the number of generator indices is $(length(indices)), " *
+                                                     "but it was expected to be $(dim(Z))"))
```

Comment:
We can keep it like that but there are three different types of errors in this PR (`throw(ArgumentError(...))`, `error(...)`, `@assert`). `@assert` is possibly the best for checking input consistency because it can be deactivated to get some extra speed.

Reply:
thanks that's true and feel free to open a follow up issue to cleanup such use cases here and other parts of the library. i think that throwing an argument error is better than an assertion; in either case, we should be able to turn these on and off using a macro  / keyword check.